src/plugins/intel_gpu/src/graph/crop.cpp

```diff
@@ -21,23 +25,125 @@ layout crop_inst::calc_output_layout(crop_node const& node, kernel_impl_params c
     auto desc = impl_param.typed_desc<crop>();
     const auto& ref_in_sizes = desc->reference_input;
     const auto in_layout = impl_param.get_input_layout();
-    const auto& in_sizes = in_layout.get_tensor();
-    const auto& offsets = desc->offsets;
+    if (in_layout.is_static()) {
+        const auto& in_sizes = in_layout.get_tensor();
+        const auto& offsets = desc->offsets;
 
-    // Check for borders variant of crop.
-    if (ref_in_sizes.batch[0] < 0 || ref_in_sizes.feature[0] < 0 || ref_in_sizes.spatial[0] < 0 ||
-        ref_in_sizes.spatial[1] < 0 || ref_in_sizes.spatial[2] < 0) {
-        // Ignore not supported dimensions.
-        const auto rb_sizes = ref_in_sizes.negate().sub({0, 0, 0, 0, 0});
-        const auto lt_sizes = offsets.sub({0, 0, 0, 0, 0});
+        // Check for borders variant of crop.
+        if (ref_in_sizes.batch[0] < 0 || ref_in_sizes.feature[0] < 0 || ref_in_sizes.spatial[0] < 0 ||
+            ref_in_sizes.spatial[1] < 0 || ref_in_sizes.spatial[2] < 0) {
+            // Ignore not supported dimensions.
+            const auto rb_sizes = ref_in_sizes.negate().sub({0, 0, 0, 0, 0});
+            const auto lt_sizes = offsets.sub({0, 0, 0, 0, 0});
 
-        const auto out_sizes = in_sizes - (rb_sizes + lt_sizes);
+            const auto out_sizes = in_sizes - (rb_sizes + lt_sizes);
 
-        return layout({in_layout.data_type, in_layout.format, out_sizes});
+            return layout({in_layout.data_type, in_layout.format, out_sizes});
+        }
```

Comment:
BTW, should we need to update this API? this old api is only to be used for static case.

Reply:
Fixed it.